gammapy/modeling/models/core.py

```diff
@@ -791,6 +791,30 @@ def reassign(self, dataset_name, new_dataset_name):
         models = [m.reassign(dataset_name, new_dataset_name) for m in self]
         return self.__class__(models)
 
+    def to_template_spatial_model(self, geom, spectral_model=None, name=None):
+        """Merge a list of models into a single `~gammapy.modeling.models.SkyModel`
+    
+        Parameters
+        ----------
+        spectral_model : `~gammapy.modeling.models.SpectralModel`
+            One of the NormSpectralMdel 
+        name : str
+            Name of the new model
+                           
+        """
+        from . import SkyModel, TemplateSpatialModel, PowerLawNormSpectralModel
+
+        unit = u.Unit("1 / (cm2 s sr TeV)")
```

Comment:
Here you could simplify by just evaluation the first component separately, like:
```python3
map_ = self[0].evaluate_geom()

for m in self[1:]:
    map_ += m.evaluate_geom()
```
I think it also misses the selection on  the model type, or raising and error if a `FoVBackgroundModel` is contained.

Reply:
`map_ = self[0].evaluate_geom()`
 this return a quantity not a map

Reply:
Indeed, you are right. Could be worth changing at some point, as we have it for `.integrate_geom()` as well...